services/rpcfilters/service.go

```diff
@@ -32,19 +37,32 @@ func (s *Service) APIs() []rpc.API {
 		{
 			Namespace: "eth",
 			Version:   "1.0",
-			Service:   NewPublicAPI(s.latestBlockChangedEvent),
-			Public:    true,
+			Service: NewPublicAPI(
+				s.latestBlockChangedEvent,
+				s.transactionSentToUpstreamEvent),
+			Public: true,
 		},
 	}
 }
 
 // Start is run when a service is started.
 func (s *Service) Start(server *p2p.Server) error {
+	err := s.transactionSentToUpstreamEvent.Start()
+	if err != nil {
+		return err
+	}
 	return s.latestBlockChangedEvent.Start()
 }
 
 // Stop is run when a service is stopped.
 func (s *Service) Stop() error {
+	s.transactionSentToUpstreamEvent.Stop()
 	s.latestBlockChangedEvent.Stop()
 	return nil
 }
+
+// TriggerTransactionSentToUpstreamEvent notifies the subscribers
+// of the TransactionSentToUpstream event
+func (s *Service) TriggerTransactionSentToUpstreamEvent() {
```

Comment:
Thought this would be the cleanest interface for triggering the event.

The alternative would be to create `transactionSentToUpstreamProvider` with a single `Trigger()` method:
```
 type Service struct {
	latestBlockChangedEvent           *latestBlockChangedEvent
	transactionSentToUpstreamEvent    *transactionSentToUpstreamEvent
        TransactionSentToUpstreamProvider *transactionSentToUpstreamProvider
 }
```

and then make the following call from backend.go
```
b.rpcFilters.TransactionSentToUpstreamProvider.Trigger()
```

Reply:
The current solution looks fine.